server/pulp/server/async/scheduler.py

```diff
@@ -368,3 +369,6 @@ def add(self, **kwargs):
         entries to the database, and they will be picked up automatically.
         """
         raise NotImplementedError
+
+    def close(self):
```

Comment:
scheduler close is called by the celery beat service object when it does a sync:
https://github.com/celery/celery/blob/master/celery/beat.py#L544

By default scheduler.close just calls scheduler.sync, which does not seem to do much. Overriding it to clean up our db seems like the cleanest solution.
https://github.com/celery/celery/blob/master/celery/beat.py#L331